src/main/java/uk/gov/homeoffice/digital/sas/timecard/kafka/producers/KafkaProducerService.java

```diff
@@ -31,33 +31,40 @@ public KafkaProducerService(
   public void sendMessage(String messageKey, Class<T> resourceType,
                           T resource, KafkaAction action) {
     var kafkaEventMessage = new KafkaEventMessage<>(projectVersion, resourceType, resource, action);
-    ListenableFuture<SendResult<String, KafkaEventMessage<T>>> future =
-        kafkaTemplate.send(
-            topicName,
-            messageKey,
-            kafkaEventMessage
-        );
-
-    listenableFutureReporting(resource, kafkaEventMessage, future);
+    CompletableFuture<SendResult<String, KafkaEventMessage<T>>> future = null;
+    try {
+      future = kafkaTemplate.send(
+          topicName,
+          messageKey,
+          kafkaEventMessage
+      );
+      completeKafkaTransaction(future);
+      logKafkaMessage(messageKey, kafkaEventMessage, future);
+    } catch (ExecutionException | InterruptedException e) {
+      Thread.currentThread().interrupt();
```

Comment:
Just realised the change here we were discussing before isn't present yet, 
The one about doing the `interrupt() `only for the `InterruptException` - separate catch block / if statement

Reply:
Thought I had pushed this. It is pushed now.